src/lib/sql/sql_translator.cpp

```diff
@@ -879,6 +877,8 @@ void SQLTranslator::_translate_select_groupby_having(const hsql::SelectStatement
      */
     if (_current_lqp->find_column_id(*sub_expression)) return ExpressionVisitation::DoNotVisitArguments;
 
+    if (sub_expression->type != ExpressionType::Aggregate) return ExpressionVisitation::VisitArguments;
```

Comment:
Recurse into the arguments only if the entire expression was not already computed (line before)